### PR TITLE
[#83] Ability to edit user

### PR DESCRIPTION
**Pull request by DenisZackharov**
### Summary

A brief description of the pull request.

### How it works

Untitled (1)

### Test plan

List of steps to manually test introduced functionality:

1. Run server with bin/dev (or bin/rails s)
2. Login with credentials: `email`, `password`
3. click on your account name
4. Try to edit field
5. write current password, and click update button

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests for new functionality
- [ ] My pull request is shorter than 300 lines

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, ENV variables, services, scripts, etc.

### CODE REVIEW

File: app/views/items/_navbar.html.slim [hunk 1]
```diff
@@ -9,6 +9,9 @@ nav.navbar.navbar-expand-lg.bg-light
             =< link_to "Sign Up", new_user_registration_path,
                 class: "nav-link"
         - elsif user_signed_in?
+          li.nav-item
+            =< link_to edit_user_registration_path, class: "nav-link" do
+              = raw("#{current_user.full_name} <span class='badge badge-light'>#{current_user.role}</span>")
```
Comment (DenisZackharov) [inline app/views/items/_navbar.html.slim:14]: Maybe move to helper module

Comment (DenisZackharov) [reply app/views/items/_navbar.html.slim:14]: And fix badge color

**Comment by RasimKhusaenov** [comment]
Duplicating error message:
```
Email can't be blank and Email can't be blank
```

image

File: app/views/devise/registrations/edit.html.slim [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
 = simple_form_for(resource, as: resource_name,
   url: registration_path(resource_name),
   html: { method: :put }) do |f|
   = f.error_notification
   .form-inputs
     = f.input :email, required: true, autofocus: true
+    = f.input :first_name, autofocus: true
+    = f.input :last_name, autofocus: true
+    - if User::MANAGE_SYSTEM_ROLES.include?(current_user.role)
+      = f.input :role, autofocus: true
```
Comment (RasimKhusaenov) [inline app/views/devise/registrations/edit.html.slim:13]: Why user can update his role? And why superadmin can't?)

File: spec/features/user/destroy_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+feature "Destroy" do
+  before do
+    create(
+      :user,
+      email: "example@gmail.com",
+      password: "123456",
+      role: "director",
+      first_name: "Bilbo",
+      last_name: "Baggins"
```
Comment (RasimKhusaenov) [inline spec/features/user/destroy_spec.rb:11]: How about use trait for it?

File: spec/features/user/edit_spec.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  scenario "edit without current password" do
+    sign_in("example@gmail.com", "123456")
+    click_on "Bilbo Baggins"
+
+    fill_in "user_first_name", with: "John"
+    fill_in "user_last_name", with: "Doe"
+
+    click_on "Update"
```
Comment (RasimKhusaenov) [inline spec/features/user/edit_spec.rb:36]: Maybe move it to function? 🙈

Comment (DenisZackharov) [reply spec/features/user/edit_spec.rb:36]: @RasimKhusaenov Maybe you fuck yourself?

**Review by RasimKhusaenov — APPROVED** [review]

**Pull request merged**